fuse_optimizers/src/variable_stamp_index.cpp

```diff
@@ -47,19 +47,29 @@ namespace fuse_optimizers
 {
 ros::Time VariableStampIndex::currentStamp() const
 {
-  auto compare_stamps = [](const StampedMap::value_type& lhs, const StampedMap::value_type& rhs)
-  {
-    return lhs.second < rhs.second;
-  };
-  auto iter = std::max_element(stamped_index_.begin(), stamped_index_.end(), compare_stamps);
-  if (iter != stamped_index_.end())
-  {
-    return iter->second;
+  if(!unique_stamps_.empty()){
+    return *unique_stamps_.end();
   }
-  else
-  {
-    return ros::Time(0, 0);
+  return ros::Time(0, 0);
+}
+
+ros::Time VariableStampIndex::firstStamp() const
+{
+  if(!unique_stamps_.empty()){
+    return *unique_stamps_.begin();
   }
+  return ros::Time(0, 0);
+} 
+
+ros::Time VariableStampIndex::ithStamp(size_t idx) const{
+  if(idx < unique_stamps_.size()){
+    std::set<ros::Time>::iterator it = unique_stamps_.begin();
+    for(size_t i = 0; i < idx; i++){
+      it++;
+    }
```

Comment:
There's a function for this exact purpose
```suggestion
    std::advance(it, idx);
```